pkgs/servers/monitoring/cockpit/default.nix

```diff
@@ -88,20 +88,26 @@ stdenv.mkDerivation rec {
     udev
   ];
 
-  patches = [
-    # Instead of requiring Internet access to do an npm install to generate the package-lock.json
-    # it copies the package-lock.json already present in the node_modules folder fetched as a git
-    # submodule.
-    ./nerf-node-modules.patch
-
+  postPatch = ''
     # sysconfdir is $(prefix)/etc by default and it breaks the configuration file loading feature
     # changing sysconfdir to /etc breaks the build in this part of the makefile because it tries
     # to write to /etc inside the sandbox
     # this patch redirects it to write to $out/etc instead of /etc
-    ./fix-makefiles.patch
-  ];
+    substituteInPlace src/systemd/Makefile.am \
+      --replace 'mkdir -p $(DESTDIR)$(sysconfdir)/motd.d' 'mkdir -p $(DESTDIR)$(prefix)$(sysconfdir)/motd.d' \
+      --replace 'ln -sTfr $(DESTDIR)/run/cockpit/motd $(DESTDIR)$(sysconfdir)/motd.d/cockpit' 'ln -sTfr $(DESTDIR)$(prefix)/run/cockpit/motd $(DESTDIR)$(prefix)$(sysconfdir)/motd.d/cockpit'  \
+      --replace 'mkdir -p $(DESTDIR)$(sysconfdir)/issue.d' 'mkdir -p $(DESTDIR)$(prefix)$(sysconfdir)/issue.d' \
+      --replace 'ln -sTfr $(DESTDIR)/run/cockpit/motd $(DESTDIR)$(sysconfdir)/issue.d/cockpit.issue' 'ln -sTfr $(DESTDIR)$(prefix)/run/cockpit/motd $(DESTDIR)$(prefix)$(sysconfdir)/issue.d/cockpit.issue'
+
+    substituteInPlace src/ws/Makefile-ws.am \
+      --replace 'mkdir -p $(DESTDIR)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(sysconfdir)/cockpit/machines.d' 'mkdir -p $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/machines.d' \
+      --replace 'chmod 755 $(DESTDIR)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(sysconfdir)/cockpit/machines.d' 'chmod 755 $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/ws-certs.d $(DESTDIR)$(prefix)$(sysconfdir)/cockpit/machines.d'
```

Comment:
Can you try to upstream this?

Reply:
yes https://github.com/cockpit-project/cockpit/pull/18513

Reply:
Use this patch file you upstreamed instead of the script-patch above.

Reply:
This broke their build

I tried another approach, just editing the relevant references. Tested in my daily driver and it's working.